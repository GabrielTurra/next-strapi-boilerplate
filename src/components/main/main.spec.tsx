import { render, screen } from '@testing-library/react';
import Main from '.';

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <Main title="Next and Strapi Boilerplate" description="Boilerplate" />
    );

    expect(
      screen.getByRole('heading', { name: /Next and Strapi Boilerplate/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the colors correctly', () => {
    // renderiza o component
    const { container } = render(
      <Main title="Next and Strapi Boilerplate" description="Boilerplate" />
    );

    // verifica se o background-color est√° correto
    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' });
  });
});
