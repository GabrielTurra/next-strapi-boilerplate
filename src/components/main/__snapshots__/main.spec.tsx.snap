// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Main /> > should render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

<main
  class="c0"
>
  <h1>
    Next and Strapi Boilerplate
  </h1>
  <p>
    Boilerplate
  </p>
</main>
`;
